### PR TITLE
Task nine

### PR DESCRIPTION
**Pull request by Abrams8**

### CODE REVIEW

File: src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java [hunk 1]
```diff
@@ -4,39 +4,63 @@
...
 public class BucketController {
 
     @Autowired
     BucketService bucketService = new BucketServiceImpl();
 
     @GetMapping
-    public List<Bucket> getAllBuckets() {
-        return bucketService.getAllBuckets();
+    public ResponseEntity<List<Bucket>> getAllBuckets(@RequestHeader String role) {
+        if(role.equals("USER")){
```
Comment (ALnOUA) [inline src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java]: it`s better to flip for avoiding potential NPE
while comparing set constants on the first place
"USER".equals(role)
and make "USER" as a constant variable

File: src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java [hunk 2]
```diff
@@ -4,39 +4,63 @@
...
+    public ResponseEntity<String> clearBucket(@RequestParam int bucketId, @RequestHeader String role) {
+        if(role.equals("USER")){
+            bucketService.clearBucket(bucketId);
+            return ResponseEntity.ok("done");
+        } else {
+            return ResponseEntity.status(404).body("error");
+        }
     }
 
     @PostMapping("/makeOrder")
```
Comment (ALnOUA) [inline src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java]: dont use verbs 
https://www.partech.nl/nl/publicaties/2020/07/9-trending-best-practices-for-rest-api-development

**Review by ALnOUA — CHANGES_REQUESTED** [review]

File: src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java (hunk 1 above)
Comment (Abrams8) [reply src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java]: done

File: src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java (hunk 2 above)
Comment (Abrams8) [reply src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java]: done

**Review by ALnOUA — APPROVED** [review]

**Pull request merged**